plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearch6ConnectorTest.java

```diff
@@ -14,21 +14,61 @@
 package io.trino.plugin.elasticsearch;
 
 import com.google.common.collect.ImmutableMap;
+import io.trino.testing.QueryRunner;
+import io.trino.tpch.TpchTable;
 import org.apache.http.entity.ContentType;
 import org.apache.http.nio.entity.NStringEntity;
+import org.elasticsearch.client.RestHighLevelClient;
 import org.intellij.lang.annotations.Language;
+import org.testng.annotations.AfterClass;
 import org.testng.annotations.Test;
 
 import java.io.IOException;
 
+import static io.trino.plugin.elasticsearch.ElasticsearchQueryRunner.createElasticsearchQueryRunner;
 import static java.lang.String.format;
 
 public class TestElasticsearch6ConnectorTest
         extends BaseElasticsearchConnectorTest
 {
+    private final ElasticsearchServer elasticsearch;
+    private final RestHighLevelClient client;
+
     public TestElasticsearch6ConnectorTest()
     {
-        super("docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0");
+        try {
+            elasticsearch = new ElasticsearchServer("docker.elastic.co/elasticsearch/elasticsearch-oss:6.0.0", ImmutableMap.of());
```

Comment:
Please do not setup resources in the constructor. Testng creates all the tests instances at once, and then run tests. That way we may end up having multiple elastic search containers to be up and running while some of them are not used.

Reply:
Applied